lib/ukstore/include/uk/store.h

```diff
@@ -29,3 +29,345 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
+
+#ifndef __UK_STORE_H__
+#define __UK_STORE_H__
+
+#include <string.h>
+#include <uk/config.h>
+#include <uk/assert.h>
+#include <uk/list.h>
+#include <uk/arch/atomic.h>
+#include <uk/essentials.h>
+#include <uk/alloc.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !CONFIG_LIBUKSTORE
+#define uk_store_libid(libname) (-1)
+
+enum uk_store_entry_type;
+struct uk_store_entry;
+
+#define __UK_STORE_STATIC_ENTRY(entry, lib_str, e_type, e_get, e_set)	\
```

Comment:
hum... did you try compiling instrumented code when the library is off. Because there is no definition of the struct and datatypes, I guess this registration will fail. I remember that we added this in order to keep type checking although it is not used. In fact - we may want to fix this by making the struct and the simple types always available.

Reply:
I don't know how this got here 😅 I'm removing it now